pkg/diagnostics/diagnostic_bundle_test.go

```diff
@@ -468,19 +480,20 @@ func TestGenerateManagementClusterBundle(t *testing.T) {
 	})
 	t.Run(t.Name(), func(t *testing.T) {
 		a := givenMockAnalyzerFactory(t)
-		a.EXPECT().DefaultAnalyzers().Return(nil)
-		a.EXPECT().ManagementClusterAnalyzers().Return(nil)
-		a.EXPECT().DataCenterConfigAnalyzers(spec.Cluster.Spec.DatacenterRef).Return(nil)
+		a.EXPECT().DefaultAnalyzers().AnyTimes().Return(nil)
+		a.EXPECT().ManagementClusterAnalyzers().AnyTimes().Return(nil)
+		a.EXPECT().DataCenterConfigAnalyzers(bs_spec.Cluster.Spec.DatacenterRef).AnyTimes().Return(nil)
+		a.EXPECT().DataCenterConfigAnalyzers(wl_spec.Cluster.Spec.DatacenterRef).AnyTimes().Return(nil)
 		a.EXPECT().EksaLogTextAnalyzers(gomock.Any()).AnyTimes().Return(nil)
 
 		c := givenMockCollectorsFactory(t)
-		c.EXPECT().DefaultCollectors().Return(nil)
-		c.EXPECT().ManagementClusterCollectors().Return(nil)
-		c.EXPECT().DataCenterConfigCollectors(spec.Cluster.Spec.DatacenterRef).Return(nil)
-		c.EXPECT().APIServerCollectors(spec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host).AnyTimes().Return(nil)
+		c.EXPECT().DefaultCollectors().AnyTimes().Return(nil)
```

Comment:
Do we need AnyTimes() here for retry? If not can we limit it to One? Same below

Reply:
> Do we need AnyTimes() here for retry? If not can we limit it to One? Same below

Yep, we need retry here.

Reply:
then can you specify the `Times()` instead of `AnyTimes()`?

Reply:
Sorry about this late update. Took some time to learn the go mock and unite test. We do not need retry here and 1 time is fine. Fixed it in the recent commit.